### ISSUE TITLE
[BUG] There seems to be a problem with the implementation of the ConnectionPool

### ISSUE THREAD

**Issue by yangfanyu** [mongo-dart/mongo_dart#339]
My test code

```dart

void main() async {

  final pool = ConnectionPool(3, () => Db('mongodb://127.0.0.1:27017/shelf'));

  var db1 = await pool.connect();
  var db2 = await pool.connect();
  var db3 = await pool.connect();
  var db4 = await pool.connect();
  var db5 = await pool.connect();
  var db6 = await pool.connect();
  print(db1.hashCode); //291730507
  print(db2.hashCode); //291730507
  print(db3.hashCode); //291730507
  print(db4.hashCode); //291730507
  print(db5.hashCode); //291730507
  print(db6.hashCode); //291730507

  await pool.close();
}

```

The hashcode of the returned instance is exactly the same. 

So I checked the source code and found that the parameter _index does not seem to be incremented when creating a new Db instance at 

https://github.com/mongo-dart/mongo_dart/blob/main/lib/src/connection_pool.dart

``` dart 

class ConnectionPool {
  final List<Db> _connections = [];
  int _index = 0;
  final Pool _pool;
........
  Future<Db> connect() {
    return _pool.withResource(() async {
      var i = _index;

      /// Modify this line to "if (++_index >= maxConnections) _index = 0;"  ?   
      if (_index >= maxConnections) _index = 0;  

      if (i < _connections.length) {
        return _connections[i];
      } else {
        var db = await dbFactory();
        await db.open();
        _connections.add(db);
        return db;
      }
    });
  }
..........
}

```
If I'm mistaken please ignore this question, thanks.

**Comment by giorgiofran** [comment]
I guees that you are right. As soon as I will have time I will give it a try.

**Comment by merterkoc** [comment]
I would be glad if you use this package that I have just developed. I continue to develop and have plans to bring a lot of features.
https://pub.dev/packages/mongo_pool
https://github.com/merterkoc/mongo_pool

**Comment by giorgiofran** [comment]
Fixed in 0.9.3

**Comment by giorgiofran** [comment]
I close this issue, feel free to reopen it if needed
